src/main/clojure/clara/benchmark/accum_repeatedly_not_selected.clj

```diff
@@ -0,0 +1,24 @@
+(ns clara.benchmark.accum-repeatedly-not-selected
+  "Added to test performance improvements in clara-rules issue 182"
+  (:require [clara.rules :refer :all]
+            [clara.rules.accumulators :as acc])
+  (:import [clara.benchmark.beans Discount Customer]
+           [java.util
+            ArrayList
+            List]))
+
+(def min-id (acc/min (fn [^Customer c] (.getId c)) :returns-fact true))
+
+(defrule vip-discount
+  [?customer <- min-id :from [Customer]]
+  =>
+  (apply insert! (let [fact-list ^List (ArrayList.)]
+                   (loop [n 0]
+                     (.add fact-list (Discount. 10))
+                     (if (< n 1000)
+                       (recur (inc n))
+                       fact-list)))))
```

Comment:
The reasoning here was to avoid any possible overhead from Clojure laziness.  There was some discussion on the previous diff of this PR.  I think perhaps just using the for loop and more generally not worrying about the exact perf numbers might be the way to go here.  Some quick REPL testing suggests this ArrayList implementation might not be faster anyway.  This benchmark is a highly artificial workflow; to me the value is just in looking at the order of magnitude of the performance of Drools and Clara here, rather than the exact numbers.  I'm leaning back toward using a Clojure "for" implementation.

For context, Drools was 2-3 ms, Clara with my prototype was 30-something ms, and master in Clara was 2 - 3 seconds.

Since we discussed this before, do you have any thoughts @mrrodriguez ?